### PR TITLE
Migration to create teacher feedback table

### PR DESCRIPTION
**Pull request by epeach**
This is a preliminary PR for the teacher feedback migration - separating migration from the rest of the code.

This feature allows teachers to leave feedback on student work. Per level, teachers can see their most recent comment and submit new feedback. Per level, students can see the most recent feedback comment from each teacher who has commented on their work.

### CODE REVIEW

File: dashboard/db/migrate/20180525181121_create_teacher_feedbacks.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateTeacherFeedbacks < ActiveRecord::Migration[5.0]
+  def change
+    create_table :teacher_feedbacks do |t|
+      t.text :comment
+      t.integer :student_id
+      t.integer :level_id
+      t.integer :teacher_id
+
+      t.timestamps
+    end
```
Comment (aoby) [inline dashboard/db/migrate/20180525181121_create_teacher_feedbacks.rb:10]: What about indexes? What will be the common queries? If we're, say, typically returning all feedbacks for a given student, we should index `student_id`

**Review by aoby — APPROVED** [review]

**Review by aoby — CHANGES_REQUESTED** [review]
Actually, this is incomplete. It's missing schema changes. It's also helpful to create an empty model that will receive the annotations. Make sure you can successfully migrate and rollback this migration.

File: dashboard/db/migrate/20180525181121_create_teacher_feedbacks.rb (hunk 1 above)
Comment (epeach) [reply dashboard/db/migrate/20180525181121_create_teacher_feedbacks.rb:10]: I think our most common query will be when students look up their feedback on a level (which is looked up by student_id and level_id: https://github.com/code-dot-org/code-dot-org/pull/22789/files#diff-85c0b816fb78a84b91283bdda9798782R13) - it'll happen when a student clicks to "Feedback" tab on a level. The look-up for teachers might also be pretty popular (whenever a teacher tries to add/update/look-at feedback for one of their students): https://github.com/code-dot-org/code-dot-org/pull/22789/files#diff-85c0b816fb78a84b91283bdda9798782R6 

I'll add indices for both of these situations.

**Comment by epeach** [comment]
@aoby PTAL, thanks!

**Review by aoby — APPROVED** [review]

File: dashboard/db/migrate/20180525181121_create_teacher_feedbacks.rb (hunk 1 above)
Comment (Erin007) [reply dashboard/db/migrate/20180525181121_create_teacher_feedbacks.rb:10]: @davidsbailey Does the rationale for these indexes no longer stand?

Comment (davidsbailey) [reply dashboard/db/migrate/20180525181121_create_teacher_feedbacks.rb:10]: per offline convo, the other index takes care of optimizing this query for us: http://www.mysqltutorial.org/mysql-index/mysql-composite-index/

https://github.com/code-dot-org/code-dot-org/blob/ea9d79bb0eaf2be018d768d133f3178dd0130f9b/dashboard/app/models/teacher_feedback.rb#L22-L23

**Pull request merged**